Cython/Compiler/Cache.py

```diff
@@ -0,0 +1,190 @@
+import sys
```

Comment:
Not completely sure but `Cache.py` feels like part of the `Build` directory to me.

I do think it's better all in one file though

Reply:
> Not completely sure but Cache.py feels like part of the Build directory to me.

I was thinking where to put this file. I put it here because I want to use it in `Main.py` in next PR (#6091) and hence I fill that it should be included in 'core' of python but still not 100% sure. If you still think it should go to `Build` I don't have any issue with that. Let me know.

Reply:
Ah I see. I don't think it should go in Compiler but I think I'll defer to @scoder about where it should go. Probably leave it for now